src/twisted/test/test_twistd.py

```diff
@@ -994,7 +994,9 @@ def setgid(gid):
         self.patch(os, "setgid", setgid)
 
         options = twistd.ServerOptions()
-        options.parseOptions(["--nodaemon", "--uid", str(wantedUid)])
+        options.parseOptions(
+            ["--nodaemon", "--uid", str(wantedUid), "--pidfile={}".format(pidFile)]
```

Comment:
This is more of a workaround.

Without this `$ trial -u -j2 twisted.test.test_twistd` would fail with something like

```
Traceback (most recent call last):
  File "/home/adi/chevah/twisted/src/twisted/test/test_twistd.py", line 1012, in test_applicationStartsWithConfiguredNameIDs
    self._applicationStartsWithConfiguredID(
  File "/home/adi/chevah/twisted/src/twisted/test/test_twistd.py", line 981, in _applicationStartsWithConfiguredID
    runner.preApplication()
  File "/home/adi/chevah/twisted/src/twisted/scripts/_twistd_unix.py", line 205, in preApplication
    checkPID(self.config["pidfile"])
  File "/home/adi/chevah/twisted/src/twisted/scripts/_twistd_unix.py", line 109, in checkPID
    sys.exit(
builtins.SystemExit: Another twistd server is running, PID 290716

This could either be a previously started instance of your application or a
different application entirely. To start a new one, either run it in some other
directory, or use the --pidfile and --logfile parameters to avoid clashes.


twisted.test.test_twistd.UnixApplicationRunnerStartApplicationTests.test_applicationStartsWithConfiguredNameIDs
-------------------------------------------------------------------------------
Ran 88 tests in 0.167s
```